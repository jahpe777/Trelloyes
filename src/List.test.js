import React from 'react';
import ReactDOM from 'react-dom';
import './List.css';
import List from './List';
import renderer from 'react-test-renderer';

  it('render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<List cards = {[]}/>, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
        .create(<List header = "banana" cards = {[]} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });

  it('renders the UI as expected', () => {
    const tree = renderer
        .create(<List header = "banana" cards = {[ {id: 'a', title: 'First card', content: 'lorem ipsum'} ]} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });

 
