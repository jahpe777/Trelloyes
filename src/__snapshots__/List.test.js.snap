// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      banana
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
      <p>
        dsfsd
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`renders the UI as expected 2`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      banana
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
